src/SprykerEco/Yves/Braintree/Form/CreditCardSubForm.php

```diff
@@ -138,22 +138,25 @@ public function buildView(FormView $view, FormInterface $form, array $options)
         $view->vars[static::CLIENT_TOKEN] = $this->generateClientToken();
         $view->vars[static::IS_3D_SECURE] = (string)Config::get(BraintreeConstants::IS_3D_SECURE);
 
-        /** @var \Generated\Shared\Transfer\QuoteTransfer $quote */
-        $quote = $form->getParent()->getViewData();
-
-        $view->vars[static::EMAIL] = $quote->getCustomer()->getEmail();
-        $view->vars[static::AMOUNT] = $quote->getTotals()->getGrandTotal();
-        $view->vars[static::BILLING_ADDRESS] = [
-            static::BILLING_ADDRESS_GIVEN_NAME => $this->convertToGermanAsciiFormat($quote->getBillingAddress()->getFirstName()),
-            static::BILLING_ADDRESS_SURNAME => $this->convertToGermanAsciiFormat($quote->getBillingAddress()->getLastName()),
-            static::BILLING_ADDRESS_PHONE_NUMBER => $quote->getBillingAddress()->getPhone(),
-            static::BILLING_ADDRESS_STREET_ADDRESS => $quote->getBillingAddress()->getAddress1(),
-            static::BILLING_ADDRESS_EXTENDED_ADDRESS => $quote->getBillingAddress()->getAddress2(),
-            static::BILLING_ADDRESS_LOCALITY => $quote->getBillingAddress()->getCountry() ? $quote->getBillingAddress()->getCountry()->getName() : '',
-            static::BILLING_ADDRESS_REGION => $quote->getBillingAddress()->getRegion(),
-            static::BILLING_ADDRESS_POSTAL_CODE => $quote->getBillingAddress()->getZipCode(),
-            static::BILLING_ADDRESS_COUNTRY_CODE => $quote->getBillingAddress()->getCountry() ? $quote->getBillingAddress()->getCountry()->getIso2Code() : '',
-        ];
+        $parentForm = $form->getParent();
+        if ($parentForm instanceof FormInterface) {
+            /** @var \Generated\Shared\Transfer\QuoteTransfer $quote */
+            $quote = $parentForm->getViewData();
+
+            $view->vars[static::EMAIL] = $quote->getCustomer()->getEmail();
+            $view->vars[static::AMOUNT] = $quote->getTotals()->getGrandTotal();
+            $view->vars[static::BILLING_ADDRESS] = [
+                static::BILLING_ADDRESS_GIVEN_NAME => $this->convertToGermanAsciiFormat($quote->getBillingAddress()->getFirstNameOrFail()),
+                static::BILLING_ADDRESS_SURNAME => $this->convertToGermanAsciiFormat($quote->getBillingAddress()->getLastNameOrFail()),
+                static::BILLING_ADDRESS_PHONE_NUMBER => $quote->getBillingAddress()->getPhone(),
+                static::BILLING_ADDRESS_STREET_ADDRESS => $quote->getBillingAddress()->getAddress1(),
```

Comment:
Fixed.

Reply:
Instead of repeating the getOrFail() call over and over the whole code block can be made easier to read by doing this. Please change it:

```php
            $billingAddressTransfer = quoteTransfer->getBillingAddressOrFail();

            $view->vars[static::EMAIL] = $quoteTransfer->getCustomerOrFail()->getEmail();
            $view->vars[static::AMOUNT] = $quoteTransfer->getTotals()->getGrandTotal();
            $view->vars[static::BILLING_ADDRESS] = [
                static::BILLING_ADDRESS_GIVEN_NAME => $this->convertToGermanAsciiFormat($billingAddressTransfer->getFirstNameOrFail()),
                static::BILLING_ADDRESS_SURNAME => $this->convertToGermanAsciiFormat($billingAddressTransfer->getLastNameOrFail()),
                static::BILLING_ADDRESS_PHONE_NUMBER => $billingAddressTransfer->getPhone(),
                static::BILLING_ADDRESS_STREET_ADDRESS => $billingAddressTransfer->getAddress1(),
                static::BILLING_ADDRESS_EXTENDED_ADDRESS => $billingAddressTransfer->getAddress2(),
                static::BILLING_ADDRESS_LOCALITY => $billingAddressTransfer->getCountry() ? $billingAddressTransfer->getCountry()->getName() : '',
                static::BILLING_ADDRESS_REGION => $billingAddressTransfer->getRegion(),
                static::BILLING_ADDRESS_POSTAL_CODE => $billingAddressTransfer->getZipCode(),
                static::BILLING_ADDRESS_COUNTRY_CODE => $billingAddressTransfer->getCountry() ? $billingAddressTransfer->getCountry()->getIso2Code() : '',
```